server/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java

```diff
@@ -417,7 +417,7 @@ private void internalRecoverFromStore(IndexShard indexShard) throws IndexShardRe
                     logger.debug("failed to list file details", e);
                 }
             } else {
-                store.createEmpty();
+                store.createEmpty(indexShard.indexSettings().getIndexVersionCreated().luceneVersion);
```

Comment:
can you also adapt the logic in `addIndices` in this class so that it uses the indexmetadata created version when shrinking/splitting? We then have it uniformly everywhere